llama_index/indices/vector_store/base.py

```diff
@@ -36,19 +45,31 @@ class VectorStoreIndex(BaseIndex[IndexDict]):
     def __init__(
         self,
         nodes: Optional[Sequence[BaseNode]] = None,
-        objects: Optional[Sequence[IndexNode]] = None,
-        index_struct: Optional[IndexDict] = None,
-        service_context: Optional[ServiceContext] = None,
-        storage_context: Optional[StorageContext] = None,
+        # vector store index params
         use_async: bool = False,
         store_nodes_override: bool = False,
+        embed_model: Optional[EmbedType] = None,
         insert_batch_size: int = 2048,
+        # parent class params
+        objects: Optional[Sequence[IndexNode]] = None,
+        index_struct: Optional[IndexDict] = None,
+        storage_context: Optional[StorageContext] = None,
+        callback_manager: Optional[CallbackManager] = None,
+        transformations: Optional[List[TransformComponent]] = None,
         show_progress: bool = False,
+        # deprecated
+        service_context: Optional[ServiceContext] = None,
         **kwargs: Any,
     ) -> None:
         """Initialize params."""
         self._use_async = use_async
         self._store_nodes_override = store_nodes_override
+        self._embed_model = (
+            resolve_embed_model(embed_model)
```

Comment:
So i encountered a general issue while trying to play around with the TokenCountingHandler.ipynb notebook (in docs/examples/callbacks), and noticing that all the token counts were 0.

The core issue is that the callbacks aren't automatically passed to the downstream embedding/LLM models. Before if you didn't specify LLM/embeddings the ServiceContext would make sure the callback is passed to the default LLM/embedding (i think). Now it's not, leading to behavior like token counting being broken.

A quick way to resolve this would be to do something like 
```
self._callback_manager = (
            callback_manager
            or callback_manager_from_settings_or_context(Settings, service_context)
        )
``` 

on the base LLM and embedding

Reply:
(ideally though all resolve_embed_model and resolve_llm calls will also take in the current callback manager)

Reply:
Yea, let me take a look at this.

I really think at some point we should just make callbacks global -- the logic for passing them down is so easy to miss lol

Reply:
I think we can fix this in the `getter` on the `Settings` -- the getter should always make sure the llm/embed model has the most recent callback manager (I think?)

Reply:
should we also add a callback option in case user does want to pass it manually and still use `resolve_embed_model` and `resolve_llm`?

Reply:
@jerryjliu applied the above fix, seems to work fine

Reply:
@jerryjliu I can check the interfaces where resolve is used to see if that makes sense

Reply:
ok, double done

Reply:
nice! i see the change